### PR TITLE
[chore] Fix prepare-release.yml

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** 

Fixes https://github.com/open-telemetry/opentelemetry-collector/actions/runs/8139621131/workflow

```
The workflow is not valid. .github/workflows/prepare-release.yml (Line: 54, Col: 9): 
Job 'prepare-release' depends on unknown job 'validate-version'.
```

#9409 introduced some errors because the workflow is not tested on PRs, #9668 partly fixed them and I mistakenly believed Github would notify me of further errors in the workflow, the same way it did with this one, but it did not.

### CODE REVIEW

**Review by astencel-sumo — APPROVED** [review]

**Comment by mx-psi** [comment]
Thanks for the review Andrzej!! Can't merge yet until one of @open-telemetry/collector-approvers clicks the button...

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**
